flink-libraries/flink-gelly/src/main/java/org/apache/flink/graph/asm/translate/Translate.java

```diff
@@ -0,0 +1,346 @@
+
+/**
+ * Methods for translation of the type or modification of the data of graph
+ * labels, vertex values, and edge values.
+ */
+public class Translate {
+
+	// --------------------------------------------------------------------------------------------
+	//  Translate vertex labels
+	// --------------------------------------------------------------------------------------------
+
+	/**
+	 * Translate {@link Vertex} labels using the given {@link MapFunction}.
+	 *
+	 * @param vertices input vertices
+	 * @param translator implements conversion from {@code OLD} to {@code NEW}
+	 * @param <OLD> old vertex label type
+	 * @param <NEW> new vertex label type
+	 * @param <VV> vertex value type
+	 * @return translated vertices
+	 */
+	public static <OLD,NEW,VV> DataSet<Vertex<NEW,VV>> translateVertexLabels(DataSet<Vertex<OLD,VV>> vertices, MapFunction<OLD,NEW> translator) {
+		return translateVertexLabels(vertices, translator, PARALLELISM_UNKNOWN);
+	}
+
+	/**
+	 * Translate {@link Vertex} labels using the given {@link MapFunction}.
+	 *
+	 * @param vertices input vertices
+	 * @param translator implements conversion from {@code OLD} to {@code NEW}
+	 * @param parallelism operator parallelism
+	 * @param <OLD> old vertex label type
+	 * @param <NEW> new vertex label type
+	 * @param <VV> vertex value type
+	 * @return translated vertices
+	 */
+	@SuppressWarnings("unchecked")
+	public static <OLD,NEW,VV> DataSet<Vertex<NEW,VV>> translateVertexLabels(DataSet<Vertex<OLD,VV>> vertices, MapFunction<OLD,NEW> translator, int parallelism) {
+		Preconditions.checkNotNull(vertices);
+		Preconditions.checkNotNull(translator);
+		Preconditions.checkArgument(parallelism > 0 || parallelism == PARALLELISM_DEFAULT || parallelism == PARALLELISM_UNKNOWN,
+			"The parallelism must be greater than zero.");
+
+		Class<Vertex<NEW,VV>> vertexClass = (Class<Vertex<NEW,VV>>)(Class<? extends Vertex>) Vertex.class;
+		TypeInformation<NEW> newType = TypeExtractor.createTypeInfo(MapFunction.class, translator.getClass(), 1, null, null);
+		TypeInformation<VV> vertexValueType = ((TupleTypeInfo<Vertex<OLD,VV>>) vertices.getType()).getTypeAt(1);
+
+		TupleTypeInfo<Vertex<NEW,VV>> returnType = new TupleTypeInfo<>(vertexClass, newType, vertexValueType);
+
+		return vertices
+			.map(new TranslateVertexLabel<OLD,NEW,VV>(translator))
+			.returns(returnType)
+				.setParallelism(parallelism)
+				.name("Translate vertex labels");
+	}
+
+	/**
+	 * Translate {@link Vertex} labels using the given {@link MapFunction}.
+	 *
+	 * @param <OLD> old vertex label type
+	 * @param <NEW> new vertex label type
+	 * @param <VV> vertex value type
+	 */
+	@ForwardedFields("1")
+	private static class TranslateVertexLabel<OLD,NEW,VV>
+	extends WrappingFunction<MapFunction<OLD,NEW>>
+	implements MapFunction<Vertex<OLD,VV>, Vertex<NEW,VV>> {
+		private Vertex<NEW,VV> vertex = new Vertex<>();
+
+		public TranslateVertexLabel(MapFunction<OLD,NEW> translator) {
```

Comment:
We could. I don't see a strong reason to do this since the class visibility is already private.